tests/gtest/fake_platform.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+ *  Copyright (c) 2024, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OT_GTEST_FAKE_PLATFORM_HPP_
+#define OT_GTEST_FAKE_PLATFORM_HPP_
+
+#include <openthread/config.h>
+
+#include <map>
+#include <vector>
+
+#include <inttypes.h>
+
+#include <openthread/instance.h>
+#include <openthread/platform/radio.h>
+#include <openthread/platform/time.h>
+
+namespace ot {
+
+class FakePlatform
```

Comment:
I think this is more like a `FakeNode`since it contains the OT instance. (It's composed of OT instance and a fake platform). So I think fake node is a better name for it. But it's also fine to me to keep it as is.

Reply:
While I admit `FakeNode` may be more accurate, the `test_platform` also provided APIs for OT instance initialization. Maybe it's because `otInstance` is needed to fully initialize an OT platform. Given most functions are about the platform, do you think `FakePlatform` is acceptable?

Reply:
Yeah it's fine with me.